core/store/tx_manager.go

```diff
@@ -657,6 +657,7 @@ func (txm *EthTxManager) handleSafe(
 	minimumConfirmations := txm.config.MinOutgoingConfirmations()
 	ethBalance, linkBalance, balanceErr := txm.GetETHAndLINKBalances(tx.From)
 
+	promUpdateEthBalance(ethBalance, tx.From)
```

Comment:
imo, updating the balance for prometheus should be on every successful tx attempt sent, rather than when the tx is just marked as safe.

Reply:
Updated -> https://github.com/smartcontractkit/chainlink/pull/2432/files#diff-097fbda11b23872b028772737a10e6ccR575-R583